include/ebpf_execution_type.h

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+    typedef enum _ebpf_execution_type
+    {
+        EBPF_EXECUTION_JIT,
+        EBPF_EXECUTION_INTERPRET
+    } ebpf_execution_type_t;
```

Comment:
Curious why we need an entire header file for a single typedef.  Is there no other file that this can be combined with?

Reply:
This was earlier included in ebpf_api.h. And I had to include ebpf_api.h in any file where I only wanted ebpf_execution_type_t definition. Hence I moved it to a different header file, to reduce unnecessary includes.

Reply:
Where else is it included from?   This is ok for this PR but going forward we should combine it into some more general header rather than exploding the number of include files, I think.

Reply:
This new header is also included in rpc_interface.idl. If I try to directly include ebpf_api.h (ebpf_execution_type_t was earlier defined in ebpf_api.h), then I get following compilation errors as ebpf_api.h defines some globals:

> Error	MIDL2001	instantiation of data is illegal; you must use "extern" or "static"

Reply:
agree it should move out of ebpf_api.h.   PR #246 creates an include/ebpf_structs.h file which is probably a better name for this.